src/main/java/com/opentable/versionedconfig/GitOperations.java

```diff
@@ -79,12 +79,18 @@ private Git openRepo(final GitProperties serviceConfig, Path checkoutDir)
             LOG.info("cloning {} (branch {}) to {}", remoteIndex, cloneBranch, checkoutDir);
 
             try {
+                int duration = config.getDuration();
                 final URI uri = remotes.get(remoteIndex);
                 CloneCommand clone = Git.cloneRepository()
                         .setBare(false)
                         .setBranch(cloneBranch)
                         .setDirectory(checkoutDir.toFile())
                         .setURI(uri.toString());
+
+                if (duration != -1){
```

Comment:
I'd probably move both of these to a setTimeout(TransportCommand command, config.getDuration)

Reply:
or more likely combine with configureCredentials, which already exists.